src/VisualStudio/Core/Def/Implementation/InlineRename/InlineRenameUndoManager.cs

```diff
@@ -123,10 +123,20 @@ public void CreateStartRenameUndoTransaction(Workspace workspace, ITextBuffer su
 
             public void CreateConflictResolutionUndoTransaction(ITextBuffer subjectBuffer, Action applyEdit)
             {
-                // Replace the StartRenameSession undo entry with an identically named entry that also includes
-                // the conflict resolution edits.
                 var undoManager = this.UndoManagers[subjectBuffer].UndoManager;
-                undoManager.UndoTo(this.UndoManagers[subjectBuffer].StartRenameSessionUndoPrimitive);
+
+                try
+                {
+                    // Replace the StartRenameSession undo entry with an identically named entry that also includes
+                    // the conflict resolution edits.
+                    undoManager.UndoTo(this.UndoManagers[subjectBuffer].StartRenameSessionUndoPrimitive);
+                }
+                catch (COMException ex) when (ex.ErrorCode == VSConstants.E_UNEXPECTED)
+                {
+                    // According to the documentation, E_UNEXPECTED (0x8000FFFF) is raised when the UndoManager is disabled.
```

Comment:
When does this happen, and when it does how bad is the experience?

Reply:
Users described VS closing and restarting when this exception was thrown. Not performing this `UndoTo()` did not noticably change the IDE experience even when fully commented out.